validator/lib/validatePnts.js

```diff
@@ -181,6 +181,12 @@ function validatePnts(content) {
         return 'Feature table property BATCH_LENGTH must be less than or equal to POINTS_LENGTH.';
     }
 
+    if (defined(featureTableJson.BATCH_ID)) {
+        if (featureTableJson.BATCH_ID.length > featureTableJson.BATCH_LENGTH) {
+            return 'Feature table property BATCH_LENGTH must be greater than total number of BATCH_IDs defined.';
```

Comment:
This isn't exactly the right thing to check.

Instead you should check that each value inside the `BATCH_ID` array is less than `BATCH_LENGTH`. This is where `Cesium3DTileFeatureTable.getPropertyArray` comes in, letting you get all the values and then looping over them.